ITensorGaussianMPS/src/gmps.jl

```diff
@@ -391,50 +812,110 @@ function interleave(xs...)
+  end
+  N_up = div(size(Λ_up.data, 1),site_stride(Λ_up))
+  N_dn = div(size(Λ_dn.data, 1),site_stride(Λ_up))
   N = N_up + N_dn
   ns_up, C_up = correlation_matrix_to_gmps(
     Λ_up; eigval_cutoff=eigval_cutoff, maxblocksize=maxblocksize
   )
   ns_dn, C_dn = correlation_matrix_to_gmps(
     Λ_dn; eigval_cutoff=eigval_cutoff, maxblocksize=maxblocksize
   )
-  # map the up spins to the odd sites and the even spins to the even sites
   C_up = mapindex(n -> 2n - 1, C_up)
   C_dn = mapindex(n -> 2n, C_dn)
-  C = Circuit(interleave(C_up.rotations, C_dn.rotations))
-  ns = interleave(ns_up, ns_dn)
+  C_up_rot=set_data(Λ_up,C_up.rotations)
+  C_dn_rot=set_data(Λ_dn,C_dn.rotations)
+  ns_up=set_data(Λ_up,ns_up)
+  ns_dn=set_data(Λ_dn,ns_dn)
+  C=Circuit(interleave(C_up_rot,C_dn_rot).data)
+  ns=interleave(ns_up,ns_dn).data
   if all(hastags("Fermion"), s)
-    @assert length(s) == N
-    U = [ITensor(s, g) for g in reverse(C.rotations)]
-    ψ = MPS(s, n -> round(Int, ns[n]) + 1, U; kwargs...)
+    U = itensors(s, set_data(Λ_up, C))
+    ψ = MPS(MPS_Elt, s, n -> round(Int, ns[site_stride(Λ_up) * n]) + 1)
+    ψ = apply(U, ψ; kwargs...)
   elseif all(hastags("Electron"), s)
+    ###ToDo: Not sure what to do here yet. Just copied code from old interface below. 
     @assert length(s) == N_up
     @assert length(s) == N_dn
     if isspinful(s)
-      space_up = [QN(("Nf", 0, -1), ("Sz", 0)) => 1, QN(("Nf", 1, -1), ("Sz", 1)) => 1]
-      space_dn = [QN(("Nf", 0, -1), ("Sz", 0)) => 1, QN(("Nf", 1, -1), ("Sz", -1)) => 1]
+      if typeof(Λ_up)<:ConservesNf
+        space_up = [QN(("Nf", 0, -1), ("Sz", 0)) => 1, QN(("Nf", 1, -1), ("Sz", 1)) => 1]
+        space_dn = [QN(("Nf", 0, -1), ("Sz", 0)) => 1, QN(("Nf", 1, -1), ("Sz", -1)) => 1]
+      elseif typeof(Λ_up)<:ConservesNfParity
+        ###should probably lead to an exception since parity-only contradicts spinfulness
+        space_up = [QN(("NfParity", 0, 2),) => 1, QN(("NfParity", 1, 2),) => 1]
+        space_dn = [QN(("NfParity", 0, 2),) => 1, QN(("NfParity", 1, 2),) => 1]
+      end
       sf_up = [Index(space_up, "Fermion,Site,n=$(2n-1)") for n in 1:N_up]
       sf_dn = [Index(space_dn, "Fermion,Site,n=$(2n)") for n in 1:N_dn]
       sf = collect(Iterators.flatten(zip(sf_up, sf_dn)))
     else
-      sf = siteinds("Fermion", N; conserve_qns=true, conserve_sz=false)
+      if typeof(Λ_up)<:ConservesNf
+        sf = siteinds("Fermion", N; conserve_qns=true, conserve_sz=false)
+      elseif typeof(Λ_up)<:ConservesNfParity
+        sf = siteinds("Fermion", N; conserve_qns=false, conserve_sz=false,conserve_nfparity=true)
+      end
     end
-    U = [ITensor(sf, g) for g in reverse(C.rotations)]
-    ψf = MPS(sf, n -> round(Int, ns[n]) + 1, U; kwargs...)
+    U = itensors(sf, set_data(Λ_up, C))
+    ψf = MPS(MPS_Elt, sf, n -> round(Int, ns[site_stride(Λ_up) * n]) + 1)
+    @show norm(ψf)
+    ψf = apply(U, ψf; kwargs...)
+    @show norm(ψf)
     ψ = MPS(N_up)
     for n in 1:N_up
```

Comment:
State is normalized up to application of the combiners. Norm explodes after application of the combiners. Probably an issue with the way I defined the `space` above? Not an issue for number-conserving state.

Reply:
Did you check the indices/QN spaces of the combiners and the MPS tensors before/after the application, say for a very small MPS?

Maybe related to https://github.com/ITensor/ITensors.jl/issues/1034?

Reply:
I located the problem to be in `identity_blocks_itensor` but have no idea why it fails for non-number conserving case while it works in the number-conserving one. I split the combiner application from the identity_blocks_itensor application for testing purposes and the norm only diverges after the latter. Unlikely that it's related to #1034

Reply:
I think the mapping inside `identity_blocks_itensor` may go wrong when the dimension of the subspaces is > 1. I'll test that independently.

Reply:
`identity_blocks` is only works for blocksizes of 1 or if input and output indices have the same (compatible layout). Basically mapping between the combined `Fermion` space with only `NfParity` as quantum number and an `Electron` space with only NfParity quantum number is not a well-defined operation for two `QNIndex`.

Reply:
Yes, I could believe that the current code doesn't handle that properly and that is a bit tricky to write with current BlockSparse/QN functionality

Reply:
@b-kloss I guess these comments were related to trying to get `Electron` site types working with general Gaussian states with pairing correlations which we are leaving for future work so this thread can be resolved?